### DISCUSSION TITLE
unable to add wmf delegate package for linux imagemagick installation

### DISCUSSION THREAD

**Discussion by lalitap25** [ImageMagick/ImageMagick#6860] [Development]
### ImageMagick version

7.1.1-21

### Operating system, version and so on

alpine linux

### Description

Hi There I am trying to create imagemagick container with wmf support as our application has wmf  and emf files which needs to be converted into png . But linux installation doesn't come with wmf support so i am making the image from scratch. Here is the piece of code.

RUN apk update \
    && apk add --no-cache \
        build-base \
        libwmf \
        libwmf-dev \
        imagemagick \
        imagemagick-dev 

WORKDIR /app

COPY ImageMagick.tar.gz .

RUN tar -xzvf ImageMagick.tar.gz
RUN ls -l
WORKDIR /app/ImageMagick-7.1.1-21
RUN ./configure --with-wmf=/usr/include \
    && make \
    && make install

-------

When I connect to container, I do see libwmf at /usr/include/
.configure complains about  ipa.h file which is also present under /usr/include/libwmf 
checking for WMF support... 
checking for libwmf/ipa.h... no
checking if WMF package is complete... no

I am stuck and not sure how to include wmf in my image

Any help is highly appriciated.

**Comment by urban-warrior** [comment]
Check `config.log`.  Search for the WMF validation log records and it will tell you why the WMF delegate library does not validate.  On our system the WMF delegate library validates with this command: `./configure --with-wmf`.  Do you have `wmflite` installed?

**Comment by amirjj** [comment]
I got below packages installed on Linux, however I still got DelegationError on converting `emf `to `png`. is this even possible to convert in Linux?

[MissingDelegateError no decode delegate for this image format `EMF' converting image/x-emf to image/x-png](https://stackoverflow.com/questions/78792502/missingdelegateerror-no-decode-delegate-for-this-image-format-emf-converting-i)

```
/opt/app# apt list --installed | grep -i libwmf

libwmf-dev/oldstable,now 0.2.8.4-17 amd64 [installed,automatic]
libwmf0.2-7/oldstable,now 0.2.8.4-17 amd64 [installed,automatic]


/opt/app# apt list --installed | grep -i imagemagic

imagemagick-6-common/oldstable,oldstable-security,now 8:6.9.11.60+dfsg-1.3+deb11u1 all [installed,automatic]
```

**Comment by snibgo** [comment]
Is the format listed in `magick`?
```
magick -list format |grep -i wmf
magick -list format |grep -i emf
```
Hope for outputs like this:
```
      WMF  r--   Windows Meta File
      EMF  r--   Windows Enhanced Meta File
```
I have a faint memory that the WMF library uses Windows OS facilities.

**Reply by amirjj** [reply]
No there is no `EMF` or `WMF` delegates in Linux (using alpine on docker). is there any ready to use delegates to add?
